infra/src/main/kotlin/fi/fta/geoviite/infra/logging/LoggerExternal.kt

```diff
@@ -130,3 +133,10 @@ fun Logger.integrationCall(request: ClientRequest) {
 fun Logger.integrationCall(response: ClientResponse) {
     info("External service responded: ${response.logPrefix()} status=${response.statusCode()}")
 }
+
+fun copyThreadContextToReactiveResponseThread(): ExchangeFilterFunction {
+    return ExchangeFilterFunction { request: ClientRequest, next: ExchangeFunction ->
+        val requestThreadContext = ThreadContext.getContext()
+        next.exchange(request).doOnNext { _ -> requestThreadContext?.let { ctx -> ThreadContext.putAll(ctx) } }
+    }
+}
```

Comment:
En oikein tiedä mihin tämä funkkari kuuluisi. Meidän ulkoisten integraatioiden HTTP-kirjasto hyödyntää ns. async-rakennetta, jossa pyyntö- ja vastaus käsitellään eri jvm-säikeissä. Näin olle niille ei päädy samaa diagnostiikkakontekstia -> tämä funkkari kopio pyynnön kontekstin myös vastaussäikeelle.

Kontekstiahan siis tarvitaan esim. correlationId:n (jos taski on aloitettu HTTP-pyynnöstä), käyttäjänimen ja roolin näyttämiseen, ja uutena lisänä toki myös spanId:n.

Reply:
Pitäisikös nämä myös putsata jossain kohtaa? Jos oikein tätä luen niin tuo koskee nyt vain webclientin säikeitä (jotka lienee täysin oma poolinsa -> ei vaikuta muuhun käyttöön kuin clienttiin) ja jos tämä kerran tehdään jokaisen käsittelyn alussa, niin ainakin kontekstista löytyvät arvot ylikirjoittaa aiemmat. Jos kuitenkin pääsee käymään niin että edellisellä ajolla kontekstissa oli joku arvo jota seuraavalla käynnistäjällä ei ole (esim manuaalinen pyyntö vs skeduloitu), niin mikään ei nyt varmaan putsaa sitä pois?